### PR TITLE
Add support for JSON

### PR DESCRIPTION
**Pull request by tptrixtop**
Hello all,

I have designed some feature for working with JSON, this feature must simplify working with JSON, for now i have added support only for mysql, but if it needed i will implement it for postgress and will try do the same for other adapters)

The idea of that is to simplify updating of json document, to exclude working with json_set/json_merge/patch etc, we can just implement some mini-language something like this

`$update = new Update;
        $update->table('foo')
            ->set(
                ['data->foo.is_checked' => true]
            )
            ->where([
                'id = ?' => 1,
            ])
        ;`

so it will transform into this code
`UPDATE `foo` SET `data` = JSON_SET(`data`, '$.foo.is_checked', '1') WHERE id = '1'`.

### CODE REVIEW

**Comment by Xerkus** [comment]
Please do a rebase instead of merging upstream branches

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,5 @@
 /laminas-mkdoc-theme/
 /phpunit.xml
 /vendor/
+/.idea/
+/composer.phar
```
Comment (Xerkus) [inline .gitignore:10]: You should add those to your global git ignores instead of here

**Comment by tptrixtop** [comment]
ah sure i will send another pull request)

**Comment by Xerkus** [comment]
@tptrixtop you do not need to. do a rebase with `git rebase -i` then force push to the branch you opened this PR from. That would update this PR

Join our [Slack chat](https://laminas.dev/chat/#slack) if you would like some help and guidance.

**Comment by tptrixtop** [comment]
yeah, i just wanted to make it clear)) ahaha

**Pull request closed without merging**